api/src/main/java/io/opentelemetry/trace/SpanId.java

```diff
@@ -65,46 +47,38 @@ public static int getSize() {
   }
 
   /**
-   * Returns the invalid {@code SpanId}. All bytes are 0.
+   * Returns the length of the base16 (hex) representation of the {@code SpanId}.
    *
-   * @return the invalid {@code SpanId}.
-   * @since 0.1.0
+   * @since 0.8.0
    */
-  public static SpanId getInvalid() {
-    return INVALID;
+  public static int getBase16Length() {
+    return BASE16_SIZE;
   }
 
   /**
-   * Returns a {@code SpanId} whose representation is copied from the {@code src} beginning at the
-   * {@code srcOffset} offset.
+   * Returns the invalid {@code SpanId}. All bytes are 0.
    *
-   * @param src the buffer where the representation of the {@code SpanId} is copied.
-   * @param srcOffset the offset in the buffer where the representation of the {@code SpanId}
-   *     begins.
-   * @return a {@code SpanId} whose representation is copied from the buffer.
-   * @throws NullPointerException if {@code src} is null.
-   * @throws IndexOutOfBoundsException if {@code srcOffset+SpanId.getSize()} is greater than {@code
-   *     src.length}.
+   * @return the invalid {@code SpanId}.
    * @since 0.1.0
    */
-  public static SpanId fromBytes(byte[] src, int srcOffset) {
-    Utils.checkNotNull(src, "src");
-    return new SpanId(BigendianEncoding.longFromByteArray(src, srcOffset));
+  public static String getInvalid() {
+    return INVALID;
   }
 
-  /**
-   * Copies the byte array representations of the {@code SpanId} into the {@code dest} beginning at
-   * the {@code destOffset} offset.
-   *
-   * @param dest the destination buffer.
-   * @param destOffset the starting offset in the destination buffer.
-   * @throws NullPointerException if {@code dest} is null.
-   * @throws IndexOutOfBoundsException if {@code destOffset+SpanId.getSize()} is greater than {@code
-   *     dest.length}.
-   * @since 0.1.0
-   */
-  public void copyBytesTo(byte[] dest, int destOffset) {
-    BigendianEncoding.longToByteArray(id, dest, destOffset);
+  /** Generate a valid {@link SpanId} from the given long value. */
+  public static String fromLong(long id) {
```

Comment:
Any thoughts on moving these to reflect they're only used for ID generation, maybe `IdGeneratorUtil`, or we could make `IdGenerator` an abstract class with protected helper methods for generating IDs from longs?

Reply:
I liked having all the transformations "close to home" in the relevant helpers, but it wouldn't be hard to convince me to move them. Let's do that as a follow-up, if that's ok.

Reply:
Also, there are quite a few usages in test code at the moment, so I think I'd want to hold off for a follow-up PR, for sure.